nautobot_device_lifecycle_mgmt/migrations/0021_migrate_to_core_software_models.py

```diff
@@ -0,0 +1,264 @@
+            object_repr=f"{core_software_image.software_version.platform.name} - "
+            f"{core_software_image.software_version.version} - "
+            f"{core_software_image.image_file_name}"[:CHANGELOG_MAX_OBJECT_REPR],
+            request_id=uuid.uuid4(),
+        )
+
+
+def _migrate_content_types(apps, old_ct, new_ct):
+    ComputedField = apps.get_model("extras", "ComputedField")
+    CustomField = apps.get_model("extras", "CustomField")
+    CustomLink = apps.get_model("extras", "CustomLink")
+    ExportTemplate = apps.get_model("extras", "ExportTemplate")
+    JobButton = apps.get_model("extras", "JobButton")
+    JobHook = apps.get_model("extras", "JobHook")
+    Note = apps.get_model("extras", "Note")
+    ObjectChange = apps.get_model("extras", "ObjectChange")
+    ObjectPermission = apps.get_model("users", "ObjectPermission")
+    Relationship = apps.get_model("extras", "Relationship")
+    RelationshipAssociation = apps.get_model("extras", "RelationshipAssociation")
+    Status = apps.get_model("extras", "Status")
+    Tag = apps.get_model("extras", "Tag")
+    TaggedItem = apps.get_model("extras", "TaggedItem")
+    WebHook = apps.get_model("extras", "WebHook")
+
+    # Migrate ComputedField content type
+    ComputedField.objects.filter(content_type=old_ct).update(content_type=new_ct)
+
+    # Migrate CustomField content type
+    for cf in CustomField.objects.filter(content_types=old_ct):
+        cf.content_types.add(new_ct)
+
+    # Migrate CustomLink content type
+    CustomLink.objects.filter(content_type=old_ct).update(content_type=new_ct)
+
+    # Migrate ExportTemplate content type - skip git export templates
+    ExportTemplate.objects.filter(content_type=old_ct, owner_content_type=None).update(content_type=new_ct)
+
+    # Migrate JobButton content type
+    for job_button in JobButton.objects.filter(content_types=old_ct):
+        job_button.content_types.add(new_ct)
+
+    # Migrate JobHook content type
+    for job_hook in JobHook.objects.filter(content_types=old_ct):
+        job_hook.content_types.add(new_ct)
+
+    # Migrate Note content type
+    Note.objects.filter(assigned_object_type=old_ct).update(assigned_object_type=new_ct)
+
+    # Migrate ObjectChange content type
+    ObjectChange.objects.filter(changed_object_type=old_ct).update(changed_object_type=new_ct)
+
+    # Migrate ObjectPermission content type
+    for object_permission in ObjectPermission.objects.filter(object_types=old_ct):
+        object_permission.object_types.add(new_ct)
+
+    # Migrate Relationship content type
+    Relationship.objects.filter(source_type=old_ct).update(source_type=new_ct)
+    Relationship.objects.filter(destination_type=old_ct).update(destination_type=new_ct)
+
+    # Migration RelationshipAssociation content type
+    RelationshipAssociation.objects.filter(source_type=old_ct).update(source_type=new_ct)
+    RelationshipAssociation.objects.filter(destination_type=old_ct).update(destination_type=new_ct)
```

Comment:
Would it make sense to skip over the relationships that we know are replaced by true FK/M2M fields in the core model, or is that unnecessary optimization here?

Reply:
I was thinking we might want to delete that relationship as part of this migration since leaving it there makes the create device UI confusing

![image](https://github.com/nautobot/nautobot-app-device-lifecycle-mgmt/assets/75227981/9b32e4c2-2fc6-49f6-8cc5-bf2e726cfd07)